Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1503,12 +1503,15 @@ extension BrowserViewController: URLBarDelegate {
     func urlBar(urlBar: URLBarView, didSubmitText text: String) {
         // If we can't make a valid URL, do a search query.
         // If we still don't have a valid URL, something is broken. Give up.
+        let engine = profile.searchEngines.defaultEngine
         guard let url = URIFixup.getURL(text) ??
-                        profile.searchEngines.defaultEngine.searchURLForQuery(text) else {
+                        engine.searchURLForQuery(text) else {
             log.error("Error handling URL entry: \"\(text)\".")
             return
         }
 
+        Telemetry.recordEvent(SearchTelemetryFactory.makeEvent(engineID: engine.id, source: .URLBar))
```

Comment:
Since you're using different source vars than Android, I'd recommend [documenting that](https://dxr.mozilla.org/mozilla-central/source/toolkit/components/telemetry/docs/core-ping.rst)